src/compiler/crystal/compiler.cr

```diff
@@ -323,8 +323,9 @@ module Crystal
 
         target_machine.emit_obj_to_file llvm_mod, output_filename
       end
-
-      _, command, args = linker_command(program, [output_filename], output_filename, nil)
+      object_names = [output_filename]
+      output_filename = output_filename.sub(/#{Regex.escape(unit.object_extension)}$/, "")
```

Comment:
There's no need for a regular expression here:
```suggestion
      output_filename = output_filename.rchop(unit.object_extension)
```

Reply:
I was looking for that, and couldn't find it! Thanks for pointing it out.